docs/db/db-relations.md

```diff
@@ -0,0 +1,191 @@
+
+---
+
+The `group` table is the link between projects and a group of users. This table contains the following relations:
+
+- `eventId` that associates which group is related to which event.
+- `captainId` that associates the captain's user.
+- `objectId` that associates which group is related to which object.
+
+---
+
+The `group_user` table contains the relation between groups and users. A group can have several users and so do the users.
+
+---
+
+The `token` table stores the tokens ids from the hasura authorization variables for each user. This table has no relation between other tables.
+
+---
+
+The `record` table takes care of students records (bans). All relations in this table are with the table user.
+
+---
+
+The `transaction` table takes care of rewarding the user, by accumulating the user's **xp**. This table contains the following relations:
+
+- `userId` that represents the user rewarded
+- `eventId` that associates the event in which the user was rewarded. Example: the user can be rewarded for an exercise `/madere/piscine-go/quest-01/make-it-better` and the parent event of that exercise will be `/madere/piscine-go`.
+- `objectId` that associates the object in which the user was rewarded. Example: if a user passes an exercise (object), the reward will be associated to that object.
+
+---
+
+#### Results
+
+Results will be the accumulation of the following tables:
+
+- `audit`
+- `match`
+- `progress`
+- `result`
+
+This part of the database defines the users/students progress in the school.
+
+The `audit` table contains all information related to the audit system and it is one of the ways of obtaining results. This table contains the following relations:
+
+- `groupId` that associates the group being audited.
+- `auditorId` as the name says, it is linked to the user table and has a relation of many to one. This column will be the auditor.
+- `resultId` that associates the audit to the result. An audit can have one or no results (pending on the auditor review). While the results must have at least one audit.
+
+---
+
+The `match` table is another way of obtaining a result. This table is used in bonus exercises to match two students. The following relations are established:
+
+- `userId` this will be the user wanting the match to happen.
+- `matchId` this is a self related id, a match can be made by matching other students or it can be none. If the latter is none it means that the student is waiting for the match, otherwise it is a match.
+- `eventId` that associates the event to the current match. Example: a user can be waiting for the exercise `/madere/piscine-go/quest-01/teacher` that is located in the `/madere/piscine-go` event.
+- `objectId` that associates the object to the current match, this being the bonus exercise that the user is doing.
+
+---
+
+The `progress` table is the summary of multiple results. As the names says this is the students progress. The following relations are established:
```

Comment:
A progress can exist without results so maybe it should be described as something like this: 
"is the reflection of user's activity on specific path: registration to an event related, commitment to a group associated to this path, generation of result(s) expected to validate a progress on this path."